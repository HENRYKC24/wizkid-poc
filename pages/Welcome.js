import { StyleSheet, Text, View, Image, TouchableOpacity } from "react-native";
import React from "react";

import rotate from "../assets/images/rotate.png";

const Welcome = ({ navigation }) => {
  return (
    <View style={styles.container}>
      <View style={styles.sectionOne}>
        <Image style={styles.image} source={rotate} />
        <Text style={styles.path}>Your Path to Financial Wisdom</Text>
      </View>
      <View style={styles.sectionTwo}>
        <TouchableOpacity style={styles.touchable}>
          <Text style={styles.button}>GET STARTED</Text>
        </TouchableOpacity>
        <TouchableOpacity
          onPress={() => navigation.navigate("SignIn")}
          style={styles.touchable}
        >
          <Text style={styles.button}>LOG IN</Text>
        </TouchableOpacity>
      </View>
    </View>
  );
};

export default Welcome;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "space-between",
    alignItems: "center",
    padding: 20,
  },
  sectionOne: {
    justifyContent: "center",
    alignItems: "center",
    width: "100%",
    marginTop: 100,
  },
  image: {
    width: 165,
    height: 160,
  },
  path: {
    fontWeight: "700",
    fontSize: 36,
    textAlign: "center",
  },
  sectionTwo: {
    justifyContent: "center",
    alignItems: "center",
    width: "100%",
  },
  touchable: {
    width: "100%",
  },
  button: {
    borderWidth: 2,
    borderRadius: 6,
    alignSelf: "stretch",
    // flex: 1,
    width: "100%",
    height: 50,
    marginBottom: 51,
    flexDirection: "row",
    textAlign: "center",
    lineHeight: 50,
    fontWeight: "900",
    fontSize: 20,
  },
});
