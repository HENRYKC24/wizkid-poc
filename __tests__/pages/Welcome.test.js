import React from 'react';
import renderer from 'react-test-renderer';

import Welcome from '../../pages/Welcome';

describe('<Welcome />', () => {
  it('has 1 child', () => {
    const tree = renderer.create(<Welcome />).toJSON();
    expect(tree.children.length).toBe(2);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<Welcome />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});