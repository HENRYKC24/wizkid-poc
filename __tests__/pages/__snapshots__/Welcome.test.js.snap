// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Welcome /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "space-between",
      "padding": 20,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginTop": 100,
        "width": "100%",
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 160,
          "width": 165,
        }
      }
    />
    <Text
      style={
        Object {
          "fontSize": 36,
          "fontWeight": "700",
          "textAlign": "center",
        }
      }
    >
      Your Path to Financial Wisdom
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "stretch",
            "borderRadius": 6,
            "borderWidth": 2,
            "flexDirection": "row",
            "fontSize": 20,
            "fontWeight": "900",
            "height": 50,
            "lineHeight": 50,
            "marginBottom": 51,
            "textAlign": "center",
            "width": "100%",
          }
        }
      >
        GET STARTED
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "stretch",
            "borderRadius": 6,
            "borderWidth": 2,
            "flexDirection": "row",
            "fontSize": 20,
            "fontWeight": "900",
            "height": 50,
            "lineHeight": 50,
            "marginBottom": 51,
            "textAlign": "center",
            "width": "100%",
          }
        }
      >
        LOG IN
      </Text>
    </View>
  </View>
</View>
`;
